packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -44,41 +59,45 @@ exports.sourceNodes = async (
     )
     changesActivity.start()
 
-    const { secret, action, id, data } = webhookBody
-    if (pluginOptions.secret && pluginOptions.secret !== secret) {
-      reporter.warn(
-        `The secret in this request did not match your plugin options secret.`
-      )
-      return
-    }
-    if (action === `delete`) {
-      actions.deleteNode({ node: getNode(createNodeId(id)) })
-      reporter.log(`Deleted node: ${id}`)
-      return
-    }
+    try {
+      const { secret, action, id, data } = webhookBody
+      if (pluginOptions.secret && pluginOptions.secret !== secret) {
+        reporter.warn(
+          `The secret in this request did not match your plugin options secret.`
+        )
+        return
+      }
+      if (action === `delete`) {
+        actions.deleteNode({ node: getNode(createNodeId(id)) })
+        reporter.log(`Deleted node: ${id}`)
+        return
+      }
 
-    let nodesToUpdate = data
-    if (!Array.isArray(data)) {
-      nodesToUpdate = [data]
-    }
+      let nodesToUpdate = data
+      if (!Array.isArray(data)) {
+        nodesToUpdate = [data]
+      }
 
-    for (const nodeToUpdate of nodesToUpdate) {
-      await handleWebhookUpdate(
-        {
-          nodeToUpdate,
-          actions,
-          cache,
-          createNodeId,
-          createContentDigest,
-          getCache,
-          getNode,
-          reporter,
-          store,
-        },
-        pluginOptions
-      )
+      for (const nodeToUpdate of nodesToUpdate) {
+        await handleWebhookUpdate(
+          {
+            nodeToUpdate,
+            actions,
+            cache,
+            createNodeId,
+            createContentDigest,
+            getCache,
+            getNode,
+            reporter,
+            store,
+          },
+          pluginOptions
+        )
+      }
+    } catch (e) {
+      gracefullyRethrow(changesActivity, e)
```

Comment:
Curious why don't we use `finally` instead?

Reply:
because there are some other things I want to do other than just `end()` the activity and some of them need specific order which would be pain to replicate with `finally`